lib/matplotlib/dates.py

```diff
@@ -207,16 +207,19 @@ def _get_rc_timezone():
 
 def _to_ordinalf(dt):
     """
-    Convert :mod:`datetime` or :mod:`date` to the Gregorian date as UTC float
-    days, preserving hours, minutes, seconds and microseconds.  Return value
-    is a :func:`float`.
+    Convert :mod:`datetime`, :mod:`date` or :mod:`timedelta` to the Gregorian
+    date as UTC float days, preserving hours, minutes, seconds and
+    microseconds. Return value is a :func:`float`.
     """
     # Convert to UTC
     tzi = getattr(dt, 'tzinfo', None)
     if tzi is not None:
         dt = dt.astimezone(UTC)
         tzi = UTC
 
+    if isinstance(dt, datetime.timedelta):
```

Comment:
This seems like the wrong place to decide whether or not this is a `timedelta`. Despite both being about time, I think durations and points in time are very different things and should follow different code paths. Most of the date-handling stuff is simply not applicable to `timedelta`.